open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/handlers/SubjectAreaTermHandler.java

```diff
@@ -295,17 +387,60 @@ private void setGlossary(String userId, Term term, String methodName) throws Sub
                             oMRSAPIHelper.callOMRSDeClassifyEntity(methodName, userId, guid, deClassifyName);
                         }
                     }
+
+                    List<CategorySummary> suppliedCategories = suppliedTerm.getCategories();
+                    if (suppliedCategories==null && !isReplace) {
+                        // in the update case with null categories supplied then do not change anything.
+                    } else {
+                        Set<String> deleteCategorizationGuidSet = new HashSet<>();
```

Comment:
Could you split all of this big logic on a few methods? Otherwise this method is very big and hard to read